src/dmd/hdrgen.d

```diff
@@ -717,23 +718,23 @@ public:
     override void visit(AsmStatement s)
     {
         buf.writestring("asm { ");
-        Token* t = s.tokens;
+        TokenRange tokens = s.tokens;
         buf.level++;
-        while (t)
+        while (!tokens.empty)
```

Comment:
Can this be a `foreach` loop instead?

Reply:
I think I tried that, but the calls to tokens.peekFront won't work.  They always return the 2nd token in the original range.

Reply:
Hmm, that's a shame. The reason for using the range API is to be able to use `foreach` and algorithms from `std.algorithm`. If that doesn't work I'm not sure what's the point.